pkg/controller/nodes/task/handler.go

```diff
@@ -189,6 +195,12 @@ func (t *Handler) Setup(ctx context.Context, sCtx handler.SetupContext) error {
 		for _, tt := range p.RegisteredTaskTypes {
 			logger.Infof(ctx, "Plugin [%s] registered for TaskType [%s]", p.ID, tt)
 			t.plugins[tt] = cp
+			t.taskMetricsMap[tt] = &taskMetrics{
```

Comment:
Can you add the plugin type as well in the map key and the metric name too? We are designing how to allow multiple plugins for the same task type and choose the right one at runtime as we speak...

Reply:
I am unable to understand plugin type. Have not addressed.